pdp-info/info.go

```diff
@@ -0,0 +1,150 @@
+package info
+
+import (
+	"fmt"
+	"math/rand"
+	"strings"
+	"sync"
+	"time"
+
+	log "github.com/Sirupsen/logrus"
+)
+
+type streamObj struct {
+	stream PipService_GetRequestsStreamServer
+	err    chan error
+}
+
+type streamsPool struct {
+	streams map[string][]streamObj
+	sync.RWMutex
+}
+
+func newStreamsPool() *streamsPool {
+	return &streamsPool{
+		streams: make(map[string][]streamObj),
+	}
+}
+
+func (p *streamsPool) RegisterStream(stream PipService_GetRequestsStreamServer) chan error {
+	cerr := make(chan error, 1)
+
+	resp, err := stream.Recv()
+	if err != nil {
+		cerr <- err
+		return cerr
+	}
+
+	inAttrs := resp.GetAttributes()
+	if len(inAttrs) != 1 {
+		cerr <- fmt.Errorf("Expected exactly one attribute, got: %d", len(inAttrs))
+		return cerr
+	}
+
+	name := inAttrs[0].GetValue()
+	s := streamObj{stream: stream, err: cerr}
+
+	p.Lock()
+	defer p.Unlock()
+
+	if _, ok := p.streams[name]; !ok {
```

Comment:
make sure the name is unique for each PIP instance

Reply:
Name is unique for group of PIP. Map contains list of streams.